pkg/scheduler/eventhandlers.go

```diff
@@ -506,3 +506,26 @@ func nodeConditionsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
 func nodeSpecUnschedulableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
 	return newNode.Spec.Unschedulable != oldNode.Spec.Unschedulable && !newNode.Spec.Unschedulable
 }
+
+func buildPreEnqueueChecks(nodeInfo *framework.NodeInfo) []queue.PreEnqueueCheck {
+	// Except for the taint/toleration check, the following checks are aligned with
+	// kubelet admission logic located at: pkg/kubelet/lifecycle/predicate.go#GeneralPredicates
+	// TODO: unify the logic with kubelet and move to a shared place.
+	return []queue.PreEnqueueCheck{
+		func(pod *v1.Pod) bool { return len(noderesources.Fits(pod, nodeInfo)) == 0 },
```

Comment:
we should only check the filters that are not resolvable with preemption. So no "noderesources" nor "nodeports"

Reply:
Can you elaborate? A new/updated node may or may not have sufficient resources to fit an unschedulable pod, so why not check "noderesources" and "nodeprots"?

Reply:
But it could... so the whole scheduling cycle should happen. Maybe not true for New, but it is true for Update. Although, maybe the conditions that would allow preemption wouldn't change with a NodeUdpate. So, WDYT?

Reply:
Let me rethink this: suppose an incoming high-priority Pod requested 2 CPUs, and also one existing low-priority pod occupied 1 CPU. Let's say the node's capacity was 1 CPU, then the high-priority couldn't be scheduled, even with preemption. Now suppose the node's capacity got updated to 2 CPUs, in the current logic, filtering logic of noderesource would return false, and hence it missed the chance to be moved back - it should have had the chance to preempt the low-priority Pod. Is it what you're implying?

Reply:
It is permitted, but I believe it is not common. Feels like too niche of a use case to support. If this ever happens, we still have the force move every minute. I think we should just document this case and that we deliberately chose to ignore it.

Reply:
Added a comment.